A38-admin-interface-api.md

```diff
@@ -1,57 +1,36 @@
 Admin Interface API
 ----
 * Author(s): lidiz
-* Approver: markroth
-* Status: Draft
+* Approver: markdroth
+* Status: In-Review
 * Implemented in: C++, Golang, Java
-* Last updated: 2021-01-27
-* Discussion at: TBD
+* Last updated: 2021-02-05
+* Discussion at: https://groups.google.com/g/grpc-io/c/0B2f9ha5HoM
 
 ## Abstract
 
-This proposal describes a convenience API in each gRPC language to improve the usability of creating a gRPC server with admin services.
+This proposal describes a convenience API in each gRPC language to improve the usability of creating a gRPC server with admin services to expose states in the gRPC library.
 
 
 ## Background
 
-gRPC Admin Interface is part of the Proxyless Service Mesh Observability project which includes new gRPC admin services and a CLI tool (yet to release). The admin services are responsible for exposing states of the gRPC library via gRPC protocol, and the CLI tool will help users to better interpret the fetched data from the gRPC application. Currently, the existing admin server should at least expose two admin services:
+### Services Exposing gRPC Internal States
 
-* Channelz: the channel tracing library that collects stats about a channel’s traffic, underlying sockets, and credentials ([proto definition](https://github.com/grpc/grpc/blob/master/src/proto/grpc/channelz/channelz.proto));
-* [Client Status Discovery Service](https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/status/v3/csds.proto): the xDS config dump library that dumps the accepted xDS configuration with config status;
-
-However, the number of admin services might increase in the future. Solutions with no code changes, like env, or config files, are not desired (see rationale section below). On the other hand, as a library, gRPC would like to promote the usage of admin services. So, there is a need for an ideally one LOC to create and start an admin server.
-
-In addition, by providing new APIs, we abstract away the individual services with the concept admin services. We can also achieve **dynamic admin services selection**, for example, we **only add CSDS service if the xDS package is pulled in**. In this way, we won’t force users to take unnecessary dependencies.
-
-
-### Types of Services
+gRPC provides a set of official gRPC services that serve gRPC internal states. This proposal will focus on the ones that expose the state of the entire gRPC library, which is agnostic to which gRPC server they were bound to:
```

Comment:
I think that the material here in the "Background" section contains specifics that actually belong in the "Proposal" section, and the start of the "Proposal" section provides context that actually belongs here.

I think this section should say something like the following:

"""
There exists a class of RPC services whose purpose is to expose the overall state of all gRPC activity in a given binary.  This set of services may grow over time, and some services may want to be exposed only conditionally, to avoid adding unnecessary dependencies (e.g., CSDS should be used only when xDS is used in gRPC).

The goal of this proposal is to provide a mechanism where applications can tell gRPC to add all relevant admin services to a given server, allowing gRPC to automatically determine which specific admin services are relevant.  This way, individual applications don't need to change when the set of desired admin services changes.
"""

Then, in the "Proposal" section, we can start with something like this:

"""
Initially, the following admin services will be supported:
- [Channelz](https://github.com/grpc/proposal/blob/master/A14-channelz.md), which exposes internal state of channels, servers, and sockets.
- xDS Client Status Discovery Service (CSDS) (ADD LINK TO CSDS gRFC), which exposes the state of xDS resources requested by gRPC.

"""